### ISSUE TITLE
Support codegen creation of declarations/definitions 

### ISSUE THREAD

**Issue by evilenzo** [clangd/clangd#1776]
Recently Intellisense for vscode got an update where they supported creation of definitions/declarations for functions (source: https://devblogs.microsoft.com/cppblog/vs-code-c-extension-january-update-create-definitions-and-declarations/). This feature is so useful so I think it should be supported in clangd too

**Comment by HighCommander4** [comment]
For the other half (generating a declaration from a definition), we have https://github.com/clangd/clangd/issues/730 on file.

**Comment by evilenzo** [comment]
> For the other half (generating a declaration from a definition), we have #730 on file.

It looks like it's abandoned :(

**Comment by Coder-Ang** [comment]
> Is this feature supported?

From my usage in nvim, there is still no support

**Comment by HighCommander4** [comment]
The issue still being open is generally your first clue.

**Comment by ckandeler** [comment]
Note that this duplicates #730 and #445.

**Comment by HighCommander4** [comment]
> Note that this duplicates [#730](https://github.com/clangd/clangd/issues/730) and [#445](https://github.com/clangd/clangd/issues/445).

Good point, thanks. Let's close this one and folks interested in these features can follow the respective linked issues.
